noir-projects/noir-protocol-circuits/crates/rollup-lib/src/components.nr

```diff
@@ -349,126 +361,99 @@ fn get_tx_effects_hash_input(
     }
 
     // Now we know the number of fields appended, we can assign the first value:
-    // TX_START_PREFIX | 0 | txlen[0] txlen[1] | 0 | REVERT_CODE_PREFIX | 0 | revert_code
-    // Start prefix is "tx_start".to_field() => 8 bytes
-    let prefix_bytes = TX_START_PREFIX.to_be_bytes::<8>();
-    let length_bytes = (offset as Field).to_be_bytes::<2>();
+    let expected_revert_code =
+        generate_padded_revert_code(offset as Field, tx_effect.revert_code as Field);
     // REVERT CODE
-    assert_eq(
-        tx_effects_hash_input[0],
-        field_from_bytes(
-            array_concat(
-                prefix_bytes,
-                [
-                    0,
-                    length_bytes[0],
-                    length_bytes[1],
-                    0,
-                    REVERT_CODE_PREFIX,
-                    0,
-                    tx_effect.revert_code,
-                ],
-            ),
-            true,
-        ),
-    );
+    assert_eq(tx_effects_hash_input[0], expected_revert_code);
 
     (tx_effects_hash_input, offset)
 }
 
+fn generate_padded_revert_code(offset: Field, revert_code: Field) -> Field {
```

Comment:
Could we name this something like `generate_tx_start_field`? I think it might be misleading as it's not just the revert code

Reply:
Sure thing, done.